edgelet/aziot-edged/src/lib.rs

```diff
@@ -490,10 +494,12 @@ where
         settings.hostname().to_string().to_lowercase(),
     );
 
-    env.insert(
-        GATEWAY_HOSTNAME_KEY.to_string(),
-        parent_hostname.to_string(),
-    );
+    if parent_hostname != hostname {
```

Comment:
Use `to_lowercase()` on both just in case

Reply:
do we treat these as case invariant anywhere else in the codebase? I'd rather not introduce behavior "just in case"

Reply:
Yes - in iotedge check, we perform similar case invariance checks.